decidim-core/lib/decidim/has_file_path.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require "active_support/concern"
+
+module Decidim
+  module HasFilePath
+    extend ActiveSupport::Concern
+
+    private
+
+    def file_path
```

Comment:
We discussed this before but I think we need to slightly change the processing here:
- Only store the tempfile as long as it is needed (the current solution would leave it behind and let the underlying system handle clearing the old temp files periodically)
- Download the file in chunks not to fill RAM with the file contents if it's a large file
- Add `tempfile.flush` to ensure all chunks are flushed to the file buffer
- Don't store the path to a local variable because if the file is destroyed after the block is handled, it won't work properly

So taking all these points into account I would change this to something like:
```ruby
def process_file_locally(attribute)
  blob = send(attribute)
  if ActiveStorage::Blob.service.respond_to? :path_for
    yield ActiveStorage::Blob.service.path_for(blob.key)
  else
    begin
      tempfile = Tempfile.new
      tempfile.binmode
      blob.download { |chunk| tempfile.write(chunk) }
      tempfile.flush
      tempfile.rewind
      yield tempfile.path
    ensure
      tempfile.close!
    end
  end
end
```